src/sql/parts/disasterRecovery/restore/common/restoreServiceImpl.ts

```diff
@@ -153,9 +155,15 @@ export class RestoreDialogController implements IRestoreDialogController {
 			restoreOption.taskExecutionMode = TaskExecutionMode.executeAndScript;
 		}
 
-		this._restoreService.restore(this._ownerUri, restoreOption);
-		let restoreDialog = this._restoreDialogs[this._currentProvider];
-		restoreDialog.close();
+		this._restoreService.restore(this._ownerUri, restoreOption).then(() => {
+			let connectionProfile = this._connectionService.getConnectionProfile(this._ownerUri);
+			let activeNode = this._objectExplorerService.getObjectExplorerNode(connectionProfile);
+			this._objectExplorerService.refreshTreeNode(activeNode.getSession(), activeNode).then(() => {
+				this._objectExplorerService.getServerTreeView().refreshTree();
```

Comment:
are you refreshing the whole tree? if the connection is a server connection, then you need to refresh databases folder. I think you should handle part of this in the service